pennylane/pulse/parametrized_hamiltonian.py

```diff
@@ -44,74 +41,147 @@ class ParametrizedHamiltonian:
+        coeffs = [2., f1, f2]
+        observables =  [qml.PauliX(0), qml.PauliY(0), qml.PauliZ(0)]
+        H = qml.dot(coeffs, observables)
+
+    The resulting object can be passed parameters, and will return an :class:`~.Operator` representing the
+    ``ParametrizedHamiltonian`` with the specified parameters.
+
+    .. code-block:: python3
+
+        p1 = jnp.array([1., 1.])
+        p2 = 1.
+        params = [p1, p2]
+        >>> H(params, t=1.)
+        (2.0*(PauliX(wires=[0]))) + ((0.8414709568023682*(PauliY(wires=[0]))) + (1.0*(PauliZ(wires=[0]))))
 
     .. note::
+        To be able to compute the time evolution of the Hamiltonian with :func:`~.functions.evolve`,
+        these coefficient functions should be defined using ``jax.numpy`` rather than ``numpy``.
 
-        The parameters used in the ``ParametrizedHamiltonian`` call should have the same order
-        as the functions used to define this hamiltonian. For example, if we build a
-        ``ParametrizedHamiltonian`` that contains two functions:
+    We can also access the fixed and parametrized terms of the ``ParametrizedHamiltonian``.
+    The fixed term is an :class:`~.Operator`, while the parametrized term must be initialized with concrete
+    parameters to obtain an :class:`~.Operator`:
 
-        >>> import jax.numpy as jnp
-        >>> def f1(p, t):
-        ...     return p * t
-        >>> def f2(p, t):
-        ...     return p * jnp.sin(t)
-        >>> coeffs = [f1, f2]
-        >>> ops = [qml.PauliX(0), qml.PauliY(1)]
-        >>> H = qml.dot(coeffs, ops)
+    >>> H.H_fixed()
+    2*(PauliX(wires=[0]))
+    >>> H.H_parametrized([[1.2, 2.3], 4.5], 0.5)
+    (1.095316767692566*(PauliY(wires=[0]))) + (2.25*(PauliZ(wires=[0])))
 
-        And we call it using the following parameters:
+    .. details::
+        :title: Usage Details
 
-        >>> params = [4, 5]
-        >>> H(params, t=0.5)
-        (2.0*(PauliX(wires=[0]))) + (2.397127628326416*(PauliY(wires=[1])))
+        An alternative method for creating a ``ParametrizedHamiltonian`` is to multiply operators and callable
+        coefficients:
 
-        Internally we are computing ``f1(4, 0.5)`` and ``f2(5, 0.5)``.
+        .. code-block:: python3
 
-    **Example:**
+            def f1(p, t):
+                return jnp.sin(p[0] * t**2) + p[1]
 
-    .. code-block:: python3
+            def f2(p, t):
+                return p * jnp.cos(t)
 
-        coeffs = [2, lambda v, t: v[0] * jnp.sin(v[1] * t)]
-        observables =  [qml.PauliX(0), qml.PauliY(1)]
-        H = ParametrizedHamiltonian(coeffs, observables)
+            H = 2 * qml.PauliX(0) + f1 * qml.PauliY(0) + f2 * qml.PauliZ(0)
 
-    >>> H([jnp.ones(2)], 1.)
-    (2*(PauliX(wires=[0]))) + (0.8414710164070129*(PauliY(wires=[1])))
+        Whichever method is used for initializing a :class:`~.ParametrizedHamiltonian`, the terms defined with fixed
+        coefficients should come before parametrized terms to prevent discrepancy in the wire order.
```

Comment:
I think this should have a .. note:: before it. It's a pretty important usage detail and can get drowned out in the wall of text